### DISCUSSION TITLE
Pass weighting through with results

### DISCUSSION THREAD

**Discussion by crs-stephen** [maize-tech/laravel-searchable#3] [Ideas]
Would passing weighting through with the returned results allow a greater control over the scoring of returned answers allowing you to programatically treat search results differently based on a maximum/minimum score criteria?

**Comment by riccardodallavia** [comment]
Hi! Not sure I completely understood your proposal: are you suggesting adding the ability to define the weight of each searchable attribute each time you perform a search?

**Reply by crs-stephen** [reply]
No sorry - when returning search results you calculate the matched weight of each row and return it as part of the collection. You can then use the scored weight to determine how to handle results. For example:

I search for 'Test', I have 3 possible matches however based on the weighting each match has a different weighted score. When the collection is returned that score is passed along with the collection.

- Match 1 has a score of 80
- Match 2 has a score of 40
- Match 3 has a score of 10

Because the value of match 1 is so high, I could programatically calculate it is the 'best' match and handle the UX differently by giving a 'guaranteed' answer. If all returned scores are less than a predefined criteria no 'best' match is returned and potential answers are returned etc.

**Reply by riccardodallavia** [reply]
Oh ok, got it!

That makes sense, but the main collection returns a list of models, which should not be mapped or modified in any way.
In short, we want to return a collection as it would happen when using Larave's default ORM queries.

Another option would be adding a separate method which only returns the matched weights in a separate collection.
Would that make sense to you? If so, don't be afraid to send us a PR, we would be more than happy 😉
